plugins/modules/eks_fargate_profile.py

```diff
@@ -0,0 +1,344 @@
+    selectors = module.params['selectors']
+    tags = module.params['tags']
+    wait = module.params.get('wait')
+    fargate_profile = get_fargate_profile(client, module, name, cluster_name)
+
+    if fargate_profile:
+        changed = False
+        if set(fargate_profile['podExecutionRoleArn']) != set(role_arn):
+            module.fail_json(msg="Cannot modify Execution Role")
+        if set(fargate_profile['subnets']) != set(subnets):
+            module.fail_json(msg="Cannot modify Subnets")
+        if fargate_profile['selectors'] != selectors:
+            module.fail_json(msg="Cannot modify Selectors")
+
+        changed = validate_tags(client, module, fargate_profile)
+
+        if wait:
+            wait_until(client, module, 'fargate_profile_active', name, cluster_name)
+            fargateProfile = get_fargate_profile(client, module, name, cluster_name)
+
+        module.exit_json(changed=changed, **camel_dict_to_snake_dict(fargateProfile))
+
+    if module.check_mode:
+        module.exit_json(changed=True)
+
+    check_profiles_status(client, module, cluster_name)
+
+    try:
+        params = dict(fargateProfileName=name,
+                      podExecutionRoleArn=role_arn,
+                      subnets=subnets,
+                      clusterName=cluster_name,
+                      selectors=selectors,
+                      tags=tags
+                      )
+        fargateProfile = client.create_fargate_profile(**params)
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        module.fail_json_aws(e, msg="Couldn't create fargate profile %s" % name)
+
+    if wait:
+        wait_until(client, module, 'fargate_profile_active', name, cluster_name)
+        fargateProfile = get_fargate_profile(client, module, name, cluster_name)
+
+    module.exit_json(changed=True, **camel_dict_to_snake_dict(fargateProfile))
+
+
+def delete_fargate_profile(client, module):
+    name = module.params.get('name')
+    cluster_name = module.params['cluster_name']
+    existing = get_fargate_profile(client, module, name, cluster_name)
+    wait = module.params.get('wait')
+    if not existing:
+        module.exit_json(changed=False)
+    if not module.check_mode:
+        check_profiles_status(client, module, cluster_name)
+        try:
+            client.delete_fargate_profile(clusterName=cluster_name, fargateProfileName=name)
+        except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+            module.fail_json_aws(e, msg="Couldn't delete fargate profile %s" % name)
+
+    if wait:
+        wait_until(client, module, 'fargate_profile_deleted', name, cluster_name)
+
+    module.exit_json(changed=True)
+
+
+def get_fargate_profile(client, module, name, cluster_name):
+    try:
+        return client.describe_fargate_profile(clusterName=cluster_name, fargateProfileName=name)['fargateProfile']
+    except is_boto3_error_code('ResourceNotFoundException'):
+        return None
```

Comment:
We should probably catch also: `except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:`